mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt

```diff
@@ -286,4 +286,70 @@ class StatefulHomeTest {
     rule.onNodeWithText("test2").assertExists()
     rule.onNodeWithText("test").assertExists()
   }
+
+  @Test
+  fun given_userIsLoggedIn_when_editProfileName_then_nameShouldBeUpdated() = runTest {
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = buildStore {
+      collection("users") { document("1", ProfileDocument("1", name = "test", emoji = ":)")) }
+    }
+
+    val authFacade = AuthenticationFacade(auth, store)
+    val chessFacade = ChessFacade(auth, store)
+    val socialFacade = SocialFacade(auth, store)
+
+    authFacade.signInWithEmail("email@example.org", "password")
+    val user = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          val controller = rememberNavController()
+          ProvideFacades(authFacade, socialFacade, chessFacade) {
+            StatefulHome(
+                user = user,
+                controller = controller,
+            )
+          }
+        }
+
+    rule.onNodeWithText(strings.sectionSettings).performClick()
+    rule.onNodeWithTag("editProfileName").assertExists()
+    rule.onNodeWithTag("editProfileName").performClick()
+    rule.onAllNodesWithText("test").assertCountEquals(2)
+    rule.onAllNodesWithText("test")[1].performTextInput("test2")
```

Comment:
How can you be sure that the second node with text "test" is the right one ?

Reply:
I highly suggest you use a less-flaky way to retrieve the text input node. The `onAllNodesWithText` documentation does not give any guarantee on the ordering of the nodes matched in the semantics tree.

You could probably use a custom matcher, like this for instance :

```kotlin
rule.onNode(hasSetTextAction() and hasText("test")).performWhatever()
```

Reply:
Is done thx